ownCloudAppShared/User Interface/Confidential/SecureTextField.swift

```diff
@@ -0,0 +1,57 @@
+//
+//  SecureTextField.swift
+//  ownCloud
+//
+//  Created by Matthias Hühne on 09.12.24.
+//  Copyright © 2024 ownCloud GmbH. All rights reserved.
+//
+
+/*
+ * Copyright (C) 2024, ownCloud GmbH.
+ *
+ * This code is covered by the GNU Public License Version 3.
+ *
+ * For distribution utilizing Apple mechanisms please see https://owncloud.org/contribute/iOS-license-exception/
+ * You should have received a copy of this license along with this program. If not, see <http://www.gnu.org/licenses/gpl-3.0.en.html>.
+ *
+ */
+import UIKit
+import ownCloudApp
+
+class SecureTextField : UITextField {
+
+	override init(frame: CGRect) {
+		super.init(frame: .zero)
+		self.isSecureTextEntry = true
+		self.translatesAutoresizingMaskIntoConstraints = false
+	}
+	
+	required init?(coder: NSCoder) {
+		fatalError("init(coder:) has not been implemented")
+	}
+	
+	var secureContainerView: UIView {
+		guard !ConfidentialManager.shared.allowScreenshots else {
+			let view = UIView()
+			view.translatesAutoresizingMaskIntoConstraints = false
+			
+			return view
+		}
+		
+		if let secureView = self.subviews.filter({ subview in
+			type(of: subview).description().contains("CanvasView")
+		}).first {
+			secureView.translatesAutoresizingMaskIntoConstraints = false
+			secureView.isUserInteractionEnabled = true
+			return secureView
+		}
+		
+		let view = UIView()
```

Comment:
If this line of code is reached, no screenshot-protecting CanvasView could be found and there will not be any screenshot protection.

Returning a `UIView` here will keep the view structure intact, but nothing will prevent a screenshot from being taken.

Something to discuss: should the app have this fallback and continue to work even if the protection no longer is in place? Should the app crash? Should that decision be determined by us (and hardcoded) - or be configurable?